clang/lib/CIR/CodeGen/CIRGenCXX.cpp

```diff
@@ -310,47 +310,109 @@ void CIRGenModule::codegenGlobalInitCxxStructor(const VarDecl *D,
                                                 mlir::cir::GlobalOp Addr,
                                                 bool NeedsCtor, bool NeedsDtor,
                                                 bool isCstStorage) {
-  assert(D && " Expected a global declaration!");
-  CIRGenFunction CGF{*this, builder, true};
-  CurCGF = &CGF;
-  CurCGF->CurFn = Addr;
-  Addr.setAstAttr(mlir::cir::ASTVarDeclAttr::get(builder.getContext(), D));
-
-  if (NeedsCtor) {
-    mlir::OpBuilder::InsertionGuard guard(builder);
-    auto block = builder.createBlock(&Addr.getCtorRegion());
-    CIRGenFunction::LexicalScope lexScope{*CurCGF, Addr.getLoc(),
-                                          builder.getInsertionBlock()};
-    lexScope.setAsGlobalInit();
+  llvm_unreachable("NYI");
```

Comment:
This code was hoisted up to it's caller. It was either that or hoist most of the caller's code down to this. Need to remove.

Reply:
Thanks for cleaning this up, I wasn't a big fan of the latest approach